caldav/caldav.go

```diff
@@ -54,7 +54,7 @@ func ValidateCalendarObject(cal *ical.Calendar) (eventType string, uid string, e
 		if uid == "" {
 			uid = compUID
 		}
-		if uid != compUID {
+		if comp.Name == ical.CompEvent && uid != compUID {
```

Comment:
Can we check `comp.Name != ical.CompTimezone` instead? The restriction should still be applied to e.g. VTODO, VJOURNAL, and so on.